src/rpc/net.cpp

```diff
@@ -203,14 +203,14 @@ static RPCHelpMan getpeerinfo()
         obj.pushKV("bytesrecv", stats.nRecvBytes);
         obj.pushKV("conntime", stats.nTimeConnected);
         obj.pushKV("timeoffset", stats.nTimeOffset);
-        if (stats.m_ping_usec > 0) {
-            obj.pushKV("pingtime", ((double)stats.m_ping_usec) / 1e6);
+        if (stats.m_ping_time > 0us) {
+            obj.pushKV("pingtime", count_microseconds(stats.m_ping_time) * 0.000001);
```

Comment:
Maybe use this to get rid of the hardcoded constant:
```suggestion
            obj.pushKV("pingtime", std::chrono::duration<double>{stats.m_ping_time}.count());
```

Reply:
That'd be a breaking RPC change if someone changed the type of `m_ping_time` to nanoseconds. Unlikely here, but plausible for other durations, so I'd rather not encourage such code.

Reply:
```cpp
    auto us = 500000us;
    auto ns = 500000000ns;

    std::cout << us.count() * 0.000001 << std::endl;
    std::cout << std::chrono::duration<double>{us}.count() << std::endl;
    std::cout << std::chrono::duration<double>{ns}.count() << std::endl;
```
all 3 print `0.5`.

Reply:
Hmm, then it must be broken in another way. There is no way the compiler can figure out what time unit `double` should represent in the Univalue object.

Reply:
Oh, it will always represent the result in seconds? Seems fair enough, but it would be good to wrap that into a helper method, so that the C++ docs don't have to be looked up every time this is used. (This relies on the default ration being `1`)

Reply:
> all 3 print `0.5`.

yes, the only difference I see should be that `count_microseconds` won't accept a nanoseconds argument and you'd get a compile error in that case. Also the `round()` should be more correct than `+ 0.5` for negative durations, but that shouldn't matter here

> Oh, it will always represent the result in seconds

yes, std::chrono duration's default to `std::ratio<1>` which is seconds

Reply:
no millis, no nanos, no kilos, no megas, just seconds :)

Reply:
What about adding `SecondsFloat`, similar to the ` std::chrono::seconds` helper type?

```cpp
using SecondsFloat = std::chrono::duration<double>;
inline auto count_seconds_f(SecondsFloat t) { return t.count(); }

Reply:
I like the idea. Shouldn't it be `CountSeconds()` to follow the naming convention? I think `Float` in the name could be confusing because it is using `double`, not `float`.

Reply:
I'd personally prefer just `std::chrono::duration<double>`, but most likely because I've recently done a lot of chrono related refactoring at work so I'm already used to what this means :smile:

Reply:
Thanks for teaching me through code review! I essentially used @MarcoFalke's approach, but made the ratio and the return type explicit for the reader.